[HttpFoundation] Fix typo for ParameterBag getters

rendler-denis opened the pull request:
[HttpFoundation] Fix typo for max number of arguments for ParameterBag getters

| Q | A |
| --- | --- |
| Doc fix? | yes |
| New docs? | no |
| Applies to | 2.7 |
| Fixed tickets | ~ |

rendler-denis committed:
[HttpFoundation] Fix typo for ParameterBag getters

[HttpFoundation] Fix typo for max number of arguments for ParameterBag getters

xabbuh commented on components/http_foundation/introduction.rst:
```diff
@@ -140,7 +140,7 @@ has some methods to filter the input values:
     Filters the parameter by using the PHP :phpfunction:`filter_var` function.
 
-All getters takes up to three arguments: the first one is the parameter name
+All getters take up to two arguments: the first one is the parameter name
```
Well, in Symfony 2.7 you can still pass a third argument to perform a "deep search" (see the following paragraph). However, talking about only two parameters is true when it comes to Symfony 3.0.

Would you be able to make that change here and open another pull request for the `3.0` branch?

rendler-denis replied:
Hello, @xabbuh! 

Yep, you are right. I only checked against the 3.0 branch.
I will do a pull request against the 3.0 branch asap.

Thanks,
Denis

xabbuh replied:
@rendler-denis Changing "takes" to take" for all other branches indeed is right. So we can merge this one when you change back "two" to "three". :)

rendler-denis replied:
@xabbuh, I've reverted the 'two' to 'three' text and also added a description and example for the third argument. The example is based on the comment from the code and I haven't had time to check if it is 100% correct. 
Let me know if it needs any further changes.
I will do a PR for the 3.0 branch in a few hours as well.

Thanks,
Denis

xabbuh replied:
I think the addition for the third parameter is not needed (there already is a paragraph below that explains why the argument is there). I also wouldn't want to add more details about a feature of which we know that it is not there anymore in newer Symfony versions.

xabbuh replied:
So to be honest, I would rather favour to remove the description of that feature than to document it even more (but keeping it the way it is is okay for me too).

rendler-denis replied:
@xabbuh, I see your point. I reverted the text to its original state plus removal of the unnecessary 's' character.

rendler-denis replied:
@xabbuh, I've also added a PR for the 3.0 branch. #6286

Thanks,
Denis

rendler-denis committed:
[HttpFoundation]Add missing doc for ParameterBag getter

- Add missing doc for optional $deep ParameterBag getter
- Set correct number of arguments back for versions up to 2.7

rendler-denis committed:
[HttpFoundation]Revert the third param description

Remove the unnecessary description for ParameterBag::get() third argument

xabbuh commented:
Thank you @rendler-denis. I have merged your PR into the `2.3` branch. That's why your PR is shown as closed instead of merged. But you will still get the full credit for your contribution.
